src/filters/authfilter.bal

```diff
@@ -0,0 +1,196 @@
+            var result = caller->respond(errorsResponse);
+            handleError(result);
+            return false;
+        }
+
+        string? authToken = authCookie[0].value;
+        if (authToken is string) {
+            http:Request req = new;
+            req.addHeader("Cookie", "Authorization=".concat(authToken));
+            var response = self.clientEndpoint->get(config:getAsString("mosip.auth.validate_token_url"), req);
+
+            if (response is http:Response) {
+                int statusCode = response.statusCode;
+                if (statusCode != 200) {
+                    errorsResponse.statusCode = statusCode;
+                    errorsResponse.setPayload("Error in auth server " + response.toString());
+                    var result = caller->respond(errorsResponse);
+                    handleError(result);
+                    return false;
+                }
+
+                json|http:ClientError responseJson = response.getJsonPayload();
+                if (responseJson is json) {
+                    map<json> resWrapper = <map<json>>responseJson;
+                    if (!(resWrapper["errors"] is ())) {
+                        map<json> errors = <map<json>>resWrapper["errors"];
+                        errorsResponse.statusCode = 200;
+                        errorsResponse.setPayload("Internal errors in auth service");
+                        var result = caller->respond(errorsResponse);
+                        handleError(result);
+                        return false;
+                    }
+
+                    map<json> res = <map<json>>resWrapper["response"];
+                    string roles = res["role"].toString();
+                    string[] rolesArray = stringutils:split(roles, ",");
+
+                    if (role.indexOf(PUBLISH_ROLE_PREFIX, 0) is int) {
+                        return authorizePublisher(partnerID, role, rolesArray,caller);
+                    } else {
+                        return authorizeSubscriber(partnerID, role, rolesArray, res["userId"].toString(),caller);
+                    }
+
+                }
+            } else {
+                errorsResponse.statusCode = 500;
+                errorsResponse.setPayload("Error calling auth server " + response.toString());
+                var result = caller->respond(errorsResponse);
+                handleError(result);
+                return false;
+            }
+
+        } else {
+            errorsResponse.statusCode = 401;
+            errorsResponse.setPayload("Authentication token parsing error");
+            var result = caller->respond(errorsResponse);
+            handleError(result);
+            return false;
+        }
+        return false;
+    }
+};
+
+function handleError(error? result) {
+    if (result is error) {
+        log:printError(result.detail().toString());
+    }
+}
+
+function authorizePublisher(string? partnerID, string role, string[] rolesArray,http:Caller caller) returns boolean {
```

Comment:
rename this and subcribder method to start with isAuthorized, since we are checking the authorization and not giving authorization